### DISCUSSION TITLE
Need to display the current year on footer

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#10653] [Q&A]
![dateIssue](https://user-images.githubusercontent.com/5202291/140641011-2604cc92-a66a-47e2-88df-47f937d5e65a.png)

**Comment by hishamco** [comment]
Why you didn't use Liquid or Razor instead?

**Comment by saleemhaq** [comment]
I have a background of asp.net mvc and razor around couple of experiences i
have thats why please guide me how i achieve this with razor? then with
razor i able to used with my controllers in mvc thats why?

**Reply by hishamco** [reply]
If you are using Razor, simply use @DateTime.Now.Year

**Reply by saleemhaq** [reply]
still not print the date i put inside of my widget on admin side: © @DateTime.Now.Year All rights reserved. It not printing the date?

**Reply by hishamco** [reply]
Be sure you are using Razor not Liquid in this case

**Reply by saleemhaq** [reply]
So how i achieved? please share the syntax?

**Reply by hishamco** [reply]
I already did that ;)

**Reply by hishamco** [reply]
Please check our docs https://docs.orchardcore.net/en/latest/docs/reference/modules/Widgets/

**Reply by saleemhaq** [reply]
is it possible to share syntax unable to achieve :( ?

**Reply by ns8482e** [reply]
from admin you can only use liquid - However you prefer to use razor you can use razor (`.cshtml`) in your project.

To learn liquid go here - https://shopify.github.io/liquid/

**Comment by ns8482e** [comment]
`<script>` are not allowed due to `XSS` - https://docs.orchardcore.net/en/latest/docs/reference/core/Sanitizer/
however you can configure to be less restrictive depennding on your need

**Reply by saleemhaq** [reply]
> `<script>` are not allowed due to `XSS` - https://docs.orchardcore.net/en/latest/docs/reference/core/Sanitizer/ however you can configure to be less restrictive depennding on your need

share with me the link of github repo of CMS that has end to end working pages based on orchard decouple CMS if possible thanks in advance for help.

**Comment by Skrypt** [comment]
In Liquid : 

```liquid
{{ "now" | date: "%Y" }}
```

**Reply by saleemhaq** [reply]
> In Liquid :
> 
> ```
> {{ "now" | date: "%Y" }}
> ```

share with me the link of github repo of CMS that has end to end working pages based on orchard decouple CMS if possible thanks in advance for help.

**Reply by Skrypt** [reply]
We don't have any. Just documentation and videos. But that's pretty straightforward. We focus most of our effort on full CMS development.

**Comment by saleemhaq** [comment]
Can i used razor and liquid on same file is this possible? or any possibility we have in which cases please help?

**Reply by hishamco** [reply]
Both are different template languages

**Reply by Skrypt** [reply]
You can parse liquid inside a Razor view by injecting the service but you can't use Razor syntax inside a liquid template and vice versa.

**Comment by saleemhaq** [comment]
How is injecting the service so I am able to use the liquid? Can you share
with me a guide how I achieved this? or share with me a tutorial.?

**Reply by hishamco** [reply]
Using @inject, @saleemhaq I think you need to read a lot about ASP.NET Core and Razor Pages, coz many of your question are related to ASP.NET Core. Don't get me wrong, reading docs will help you a lot

All the best

**Comment by saleemhaq** [comment]
I mean which type of service I inject to use liquid inside the razor bro,
hope you got my point?

**Reply by ns8482e** [reply]
You need this https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#liquidtohtmlasync
Invest some time to read the docs

The search in docs is very comprehensive you will get most of you answers there without opening new discussion
